website/src/components/Visualisations/IIGOPayments/Util/ProcessedPaymentsData.ts

```diff
@@ -0,0 +1,116 @@
+                                team
+                            )
+                            if (current) {
+                                current.actualAlloc +=
+                                    teamAction.Pairs[0].Values[0]
+                                current.expectedAlloc +=
+                                    teamAction.Pairs[1].Values[0]
+                                TaxInfo.set(team, current)
+                            }
+                        }
+                        break
+                    case 'IslandTaxContribution':
+                        {
+                            const team =
+                                TeamName[
+                                    teamAction.ClientID as keyof typeof TeamName
+                                ]
+                            const current: TaxData | undefined = TaxInfo.get(
+                                team
+                            )
+                            if (current) {
+                                current.actualTax +=
+                                    teamAction.Pairs[0].Values[0]
+                                current.expectedTax +=
+                                    teamAction.Pairs[1].Values[0]
+                                TaxInfo.set(team, current)
+                            }
+                        }
+                        break
+                    case 'SanctionPaid':
+                        {
+                            const team =
+                                TeamName[
+                                    teamAction.ClientID as keyof typeof TeamName
+                                ]
+                            const current: TaxData | undefined = TaxInfo.get(
+                                team
+                            )
+                            if (teamAction.Pairs[0].Values[0] !== 0) {
+                                console.log(turnNumber, teamAction)
+                            }
+                            if (current) {
+                                current.actualSanction +=
+                                    teamAction.Pairs[0].Values[0]
+                                current.expectedSanction +=
+                                    teamAction.Pairs[1].Values[0]
+                                TaxInfo.set(team, current)
+                            }
+                        }
+                        break
+                    default:
+                        break
+                }
+            })
+        }
+    })
+    console.log(TaxInfo)
+
+    TaxInfo.forEach((value, key) => {
+        retData.push({
+            name: key,
+            actualTax: value.actualTax,
+            expectedTax: value.expectedTax,
+            actualAlloc: value.actualAlloc,
+            expectedAlloc: value.expectedAlloc,
+            actualSanction: value.actualSanction,
+            expectedSanction: value.expectedSanction,
+        })
+    })
+
+    return retData
```

Comment:
```
return TaxInfo.map((value, key) => {
    ....
}
```

Reply:
I dont know if i did something wrong but when i tried `TaxInfo,map` i got the error saying map function doesnt exist on type 'Map' ..?

Reply:
ah, nevermind. it's a map. you'd need to use the spread operator or something to do it. anyway--it works, and I don't want to block this.